apps/OpenSpace/ext/launcher/src/settingsdialog.cpp

```diff
@@ -0,0 +1,407 @@
+/*****************************************************************************************
+ *                                                                                       *
+ * OpenSpace                                                                             *
+ *                                                                                       *
+ * Copyright (c) 2014-2023                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include "settingsdialog.h"
+
+#include "profile/line.h"
+
+#include <QCheckBox>
+#include <QComboBox>
+#include <QDialogButtonBox>
+#include <QGridLayout>
+#include <QLabel>
+#include <QLineEdit>
+#include <QMessageBox>
+#include <QPushButton>
+
+SettingsDialog::SettingsDialog(openspace::Settings settings,
+                               QWidget* parent)
+    : QDialog(parent)
+    , _startEdit(settings)
+    , _currentEdit(settings)
+{
+    setWindowTitle("Settings");
+    createWidgets();
+    loadFromSettings(settings);
+}
+
+void SettingsDialog::createWidgets() {
+    // Layout of this dialog:
+    //
+    // -------------------------------------------------------
```

Comment:
I like this, but is it really necessary?  There might be a risk of it being outdated relatively quickly

Reply:
It's quite useful when adding things to see the numbers quickly.  We have it in a number of other places as well and really helps undetsand the layouts imo:

 - https://github.com/OpenSpace/OpenSpace/blob/master/apps/OpenSpace/ext/launcher/src/sgctedit/windowcontrol.cpp#L106
 - https://github.com/OpenSpace/OpenSpace/blob/master/apps/OpenSpace/ext/launcher/src/profile/scriptlogdialog.cpp#L55
 - https://github.com/OpenSpace/OpenSpace/blob/master/apps/OpenSpace/ext/launcher/src/profile/actiondialog.cpp#L78